src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackProcessor.java

```diff
@@ -75,16 +78,29 @@ private void attachCase(String exceptionRecordJurisdiction,
                             List<Map<String, Object>> exceptionDocuments) {
         CcdAuthenticator authenticator = authFactory.createForJurisdiction(exceptionRecordJurisdiction);
         CaseDetails theCase = ccdApi.getCase(caseRef, authenticator);
+
+        //TODO check for missing scannedDocs else empty list ?
+        List<Map<String, Object>> existingDocuments = getDocuments(theCase);
+
+        //TODO assert that there is a document number on exception record ?
+        checkForDuplicateAttachment(exceptionDocuments, caseRef, existingDocuments);
+
+        //This is done so exception record does not change state if there is a document error
         StartEventResponse event = ccdApi.startAttachScannedDocs(caseRef, authenticator, theCase);
         ccdApi.attachExceptionRecord(caseRef,
             authenticator,
             theCase,
-            insertNewScannedDocument(exceptionDocuments, theCase.getData()),
+            insertNewScannedDocument(exceptionDocuments, existingDocuments),
             event.getEventId(),
             event.getToken()
         );
     }
 
+    @SuppressWarnings("unchecked")
+    private List<Map<String, Object>> getDocuments(CaseDetails theCase) {
+        return (List<Map<String, Object>>) theCase.getData().get(SCANNED_DOCUMENTS);
```

Comment:
scanned documents should be list of scanneddoc objects and not a map.

Reply:
At the moment they are maps and i'm not sure we can/should re-deserialize this, write to json string and deserialize into an object, due to the changeable nature of this piece of json.

Reply:
As a cleanup activity I want to wrap it into a Document container object for ease of use but this does not currently affect the functionality.